File: pocket/src/main/java/com/ivianuu/pocket/Storage.java
package com.ivianuu.pocket;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import java.util.List;

/**
 * Storage
 */
public interface Storage {

    /**
     * Inserts the value
     */
    void put(@NonNull String key, @NonNull String value);

    /**
     * Returns the value of the key or null
     */
    @Nullable
    String get(@NonNull String key);

    /**
     * Deletes the key
     */
    void delete(@NonNull String key);

    /**
     * Deletes all values
     */
    void deleteAll();

    /**
     * Returns whether the storage contains the key
     */
    boolean contains(@NonNull String key);

    /**
     * Returns all keys in the path
     */
    @NonNull
    List<String> getAllKeys();

    /**
     * Returns the count of all entries
     */
    int getCount();

}

File: pocket/src/main/java/com/ivianuu/pocket/Storage.kt
package com.ivianuu.pocket

/**
 * Storage
 */
interface Storage {

    /**
     * Inserts the [value]
     */
    fun put(key: String, value: String)

    /**
     * Returns the value of the key or null
     */
    fun get(key: String): String?

    /**
     * Deletes the key
     */
    fun delete(key: String)

    /**
     * Deletes all values
     */
    fun deleteAll()

    /**
     * Returns whether the storage contains the key
     */
    fun contains(key: String): Boolean

    /**
     * Returns all keys in the path
     */
    fun getAllKeys(): List<String>
}
